packages/react/src/components/stepper/Stepper.tsx

```diff
@@ -0,0 +1,270 @@
+   */
+  renderCustomStepCountLabel?: (step, totalNumberOfSteps) => string;
+  /**
+   * A function for rendering a custom step heading
+   */
+  renderCustomStepHeading?: (step, totalNumberOfSteps, label) => string;
+  /**
+   * The selected step number. Counting starts from 1
+   */
+  selectedStep: number;
+  /**
+   * Boolean indicating small variant usage
+   */
+  small?: boolean;
+  /**
+   * The states of the steps
+   */
+  states: StepProps['state'][];
+  /**
+   * A boolean indicating step heading variant usage
+   */
+  stepHeading?: boolean;
+  /**
+   * Step heading aria level
+   */
+  stepHeadingAriaLevel?: number;
+  /**
+   * The total number of steps
+   */
+  stepsTotal: number;
+  /**
+   * Custom theme
+   */
+  theme?: StepperCustomTheme; // Custom theme styles
+};
+
+const getStepHeading = (language: Language, step: number, totalNumberOfSteps: number, label: string) => {
+  return {
+    en: `Step ${step}/${totalNumberOfSteps}: ${label}`,
+    fi: `Vaihe ${step}/${totalNumberOfSteps}: ${label}`,
+    sv: `Steg ${step}/${totalNumberOfSteps}: ${label}`,
+  }[language];
+};
+
+export const Stepper = ({
+  labels,
+  states,
+  language = 'fi',
+  selectedStep,
+  small = false,
+  stepsTotal,
+  renderCustomStepCountLabel,
+  renderCustomStateAriaLabel,
+  customSelectedAriaLabel,
+  onStepClick,
+  stepHeading,
+  stepHeadingAriaLevel = 2,
+  headingClassName,
+  dataTestId = 'hds-stepper',
+  renderCustomStepHeading,
+  theme,
+}: StepperProps) => {
+  const stepHeadingRef = useRef(null);
+  const stepperRef = useRef(null);
+  const stepRefs = useRef([]);
+  const arrLength = labels.length;
+  const [showPreviousButton, setShowPreviousButton] = useState(false);
+  const [showNextButton, setShowNextButton] = useState(false);
+  const customThemeClass = useTheme<StepperCustomTheme>(styles.stepperContainer, theme);
+
+  if (stepRefs.current.length !== arrLength) {
+    // add or remove refs
+    stepRefs.current = Array(arrLength)
+      .fill(0)
+      .map((_, index) => stepRefs.current[index] || createRef());
+  }
+
+  useEffect(() => {
+    if (stepHeadingRef.current) {
+      stepHeadingRef.current.focus();
+    }
+
+    stepRefs.current[selectedStep - 1].current.scrollIntoView({ behavior: 'smooth', inline: 'center' });
+
+    if (stepperRef.current.scrollLeft > 5) {
+      setShowPreviousButton(true);
+    } else {
+      setShowPreviousButton(false);
+    }
+
+    if (
+      stepperRef.current.scrollWidth - (stepperRef.current.parentNode.clientWidth + stepperRef.current.scrollLeft) >
+      5
+    ) {
+      setShowNextButton(true);
+    } else {
+      setShowNextButton(false);
+    }
+  }, [selectedStep]);
+
+  return (
+    <div lang={language} className={classNames(styles.stepperContainer, customThemeClass)} data-testid={dataTestId}>
+      {showPreviousButton && (
+        <div className={classNames(styles.scrollButton, styles.scrollButtonPrevious)} aria-hidden="true">
+          <button
```

Comment:
While this button is obviously hidden from assistive techologies, users with mouse can still click it. Should it have some kind of label so assistive technologies would read something if it is clicked with a mouse?+